src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,746 @@
+              (hcievt->evtserial.evt.payload[0] == 0x01) &&
+              (hcievt->evtserial.evt.payload[1] == 0x0C) &&
+              (hcievt->evtserial.evt.payload[2] == 0xFC)) {
+            phase_bd_addr = true;
+            if (hcievt->evtserial.evt.payload[3] != 0) {
+#if defined(PRINT_IPCC_INFO)
+              printf("Error: wrong BD Addr\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+            }
+            /* rx data is no more useful : not stored in the _rxbuff */
+            break;
+          }
+          /* check the Rx event of complete the previous tx power opcode 0xFC0F */
+          if ((hcievt->evtserial.evt.evtcode == TL_BLEEVT_CC_OPCODE) &&
+              (hcievt->evtserial.evt.payload[0] == 0x01) &&
+              (hcievt->evtserial.evt.payload[1] == 0x0F) &&
+              (hcievt->evtserial.evt.payload[2] == 0xFC)) {
+            phase_tx_power = true;
+            if (hcievt->evtserial.evt.payload[3] != 0) {
+#if defined(PRINT_IPCC_INFO)
+              printf("Error: wrong Tx power\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+            }
+            /* rx data is no more useful : not stored in the _rxbuff */
+            break;
+          }
+          /* check if the reset phase is in progress (opcode is 0x0C03) */
+          if ((hcievt->evtserial.evt.evtcode == TL_BLEEVT_CC_OPCODE) &&
+              (hcievt->evtserial.evt.payload[0] == 0x01) &&
+              (hcievt->evtserial.evt.payload[1] == 0x03) &&
+              (hcievt->evtserial.evt.payload[2] == 0x0C)) {
+            phase_reset = true;
+            if (hcievt->evtserial.evt.payload[3] != 0) {
+#if defined(PRINT_IPCC_INFO)
+              printf("Error: wrong reset\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+            }
+          }
+        }
+        __disable_irq();
+        /* store received data in the _rxbuff buffer */
+        len = hcievt->evtserial.evt.plen + TL_EVT_HDR_SIZE;
+        if (len <= BLE_MODULE_SHARED_MEM_BUFFER_SIZE - _write_index) {
+          /* at the position of the _write_index */
+          memcpy((uint8_t *)&_rxbuff[_write_index], (uint8_t *)&hcievt->evtserial, len);
+          /* move index */
+          _write_index += len;
+        } else {
+          data_overflow = true;
+        }
+        __enable_irq();
+      }
+      break;
+    case TL_ACL_DATA_PKT_TYPE: {
+        TL_AclDataSerial_t *acl = &(((TL_AclDataPacket_t *)hcievt)->AclDataSerial);
+        __disable_irq();
+        len = acl->length + 5;
+        if (len <= BLE_MODULE_SHARED_MEM_BUFFER_SIZE - _write_index) {
+          memcpy((uint8_t *)&_rxbuff[_write_index], (uint8_t *)acl, len);
+          /* move index */
+          _write_index += len;
+        } else {
+          data_overflow = true;
+        }
+        __enable_irq();
+      }
+      break;
+    default:
+      /* should not happen */
+#if defined(PRINT_IPCC_INFO)
+      printf("BLE TL evt_received, wrong type:%d\r\n", hcievt->evtserial.type);
+#endif /*(PRINT_IPCC_INFO)*/
+      while (1); /* let's block to check */
```

Comment:
I don't think it should block. 
Or maybe only in case of PRINT_IPCC_INFO ?

Reply:
yes, included if the if defined(PRINT_IPCC_INFO)